src/rgw/rgw_rados.cc

```diff
@@ -3664,6 +3664,7 @@ int RGWRados::rewrite_obj(rgw::sal::Object* obj, const DoutPrefixProvider *dpp,
 
   attrset.erase(RGW_ATTR_ID_TAG);
   attrset.erase(RGW_ATTR_TAIL_TAG);
+  attrset.erase(RGW_ATTR_STORAGE_CLASS);
```

Comment:
ok, this is makes sense i guess

ideally `object rewrite` would preserve the original storage class, and support the `--storage-class=` option to override what's there, but this looks like the simplest fix